src/functions/projectConfig.ts

```diff
@@ -0,0 +1,50 @@
+import { FunctionsConfig, FunctionConfig } from "../firebaseConfig";
+import { FirebaseError } from "../error";
+
+export type NormalizedConfig = [FunctionConfig, ...FunctionConfig[]];
+export type ValidatedSingle = FunctionConfig & { source: string };
+export type ValidatedConfig = [ValidatedSingle];
+
+/**
+ * Normalize functions config to return functions config in an array form.
+ */
+export function normalize(config?: FunctionsConfig): NormalizedConfig {
+  if (!config) {
+    throw new FirebaseError("No valid functions configuration detected in firebase.json");
```

Comment:
I think I'd prefer this error to be thrown before we call this function, like in `functions-config-export.ts` so we don't have an optional param. But this shouldn't block the PR